services/QuillLMS/app/models/question.rb

```diff
@@ -149,4 +154,37 @@ def rematch_type
   private def stored_as_array(key)
     data[key].class == Array
   end
+
+  private def validate_sequences
+    return if !data.present? || !data.is_a?(Hash)
+    parse_and_validate(data['incorrectSequences'])
+    parse_and_validate(data['focusPoints'])
+  end
+
+  private def parse_and_validate(sequences)
+    return if !sequences.present?
+    if sequences.is_a?(Hash)
+      sequences.each do |key, value|
+        validate_text_and_feedback(value)
+      end
+    elsif sequences.is_a?(Array)
+      sequences.each do |value|
+        validate_text_and_feedback(value)
+      end
+    end
+  end
+
+  private def validate_text_and_feedback(value)
+    errors.add(:data, "Focus Points and Incorrect Sequences must have text and feedback.") unless value['text'].present? && value['feedback'].present?
```

Comment:
[Opt](https://www.rubydoc.info/gems/rubocop-rails/RuboCop/Cop/Rails/Blank) for `blank?` over `!present?`.  Single-line block here again.
```
  if value['text'].blank? || value['feedback'].blank?
    errors.add(:data, "Focus Points and Incorrect Sequences must have text and feedback.")
    return
  end

  value['text'].split('|||').each { |regex| validate_regex(regex) }
```